include/mata/nfa.hh

```diff
@@ -1168,6 +1186,11 @@ private:
      */
     void handle_epsilon_transitions(const StateDepthPair& state_depth_pair, const TransSymbolStates& state_transitions,
                                     std::deque<StateDepthPair>& worklist);
+
+    /**
+     * Clear initial states for all segment but the first one. Clear final states for all segments but the last one.
+     */
+    void clear_initial_final_states();
```

Comment:
I find this naming confusing, as it doesn't clear all initial and final states, nor there is something like `initial final` states. I have no idea, what would be better though. Maybe `adjust_initial_and_final_states` or `update_initial_and_final_states`?

Reply:
The function removes *inner* initial and final states. Then, something like `remove_inner_initial_final_states` could be used.

Reply:
Sounds better, and maybe add `and` :)

Reply:
Great. Will adjust accordingly in the code.